lua/wikis/valorant/MatchGroup/Input/Custom/MatchPage.lua

```diff
@@ -9,217 +9,220 @@
-			newRoundDetails[i + 1] = roundDetail
-		end
-		map.roundDetails = newRoundDetails
-	end
-	if map.teams and type(map.teams) == 'table' and map.teams[0] then
-		local newTeams = {}
-		for i, team in pairs(map.teams) do
-			newTeams[i + 1] = team
-		end
-		map.teams = newTeams
-	end
+	assert(map, mapInput.matchid .. ' could not be retrieved from API.')
 
-	-- Fix round winner
-	-- We are currently getting the side the team started the NEXT round on, but it should be THIS round.
-	-- This is a temporary fix until the API is fixed
-	if map.roundDetails and type(map.roundDetails) == 'table' then
-		Array.forEach(map.roundDetails, function(roundDetail)
-			if roundDetail.round_no % 12 == 0 or roundDetail.round_no > 24 then
-				-- In overtime, the winner is the opposite of the round number
-				-- Same with the last round of each side in normal time (12, 24)
-				roundDetail.round_winner = otherSide(roundDetail.round_winner)
-			else
-				roundDetail.round_winner = roundDetail.round_winner
-			end
-		end)
-	end
+	local shouldReverse = Logic.readBool(mapInput.reversed)
 
-	-- Temporary reverse logic until the API has this feature added
-	if Logic.readBool(mapInput.reversed) then
-		map.matchInfo.team1, map.matchInfo.team2 = map.matchInfo.team2, map.matchInfo.team1
-		map.matchInfo.t1firstside = otherSide(map.matchInfo.t1firstside)
-		map.matchInfo.o1t1firstside = otherSide(map.matchInfo.o1t1firstside)
-		Array.forEach(map.roundDetails, function(roundDetail)
-			roundDetail.winningSide = otherSide(roundDetail.winningSide)
-		end)
+	if shouldReverse then
 		map.teams[1], map.teams[2] = map.teams[2], map.teams[1]
-		map.players[1], map.players[2] = map.players[2], map.players[1]
 	end
 
 	---@cast map valorantMatchDataExtended
+	-- Attach players to their teams
+	Array.forEach(map.teams, function(team)
+		team.players = Array.filter(map.players, function(player)
+			return player.team_id == team.team_id
+		end)
+	end)
 	map.matchid = mapInput.matchid
 	map.vod = mapInput.vod
 	map.finished = true
 
 	return map
 end
 
----@param map table
+---@param map valorantMatchDataExtended|table
 ---@param opponentIndex integer
 ---@return table[]?
 function CustomMatchGroupInputMatchPage.getParticipants(map, opponentIndex)
-	if not map.players then return nil end
-	local teamPlayers = map.players[opponentIndex]
-	if not teamPlayers then return end
+	if not map.teams then return nil end
+	local team = map.teams[opponentIndex]
+	if not team then return end
 
-	local players = Array.map(teamPlayers.players, function(player)
+	return Array.map(team.players, function(player)
+		local lpdbPlayerData = player.lpdb_player
 		return {
-			player = player.riot_id,
-			agent = player.agent,
-			acs = player.acs,
-			adr = nil,
-			kast = nil,
-			hs = nil,
-			kills = player.kills,
-			deaths = player.deaths,
-			assists = player.assists,
+			player = lpdbPlayerData and lpdbPlayerData.page_name or player.game_name,
+			agent = player.character.name,
```

Comment:
Similar to maps, there's also a json document for mapping agent names - does it make sense to also parse these here?

Reply:
player.character.riot_id is the raw one, our middleware parses it I suppose